### PR TITLE
lcn兼容SpringBoot2.0.4和Spring Cloud Finchley.RELEASE

### PR DESCRIPTION
**Pull request by jorrellz**
1. Update SpringBoot version to 2.0.4.RELEASE
2. Update JDK version to 1.8
3. Update Spring Cloud version  to Finchley.RELEASE

### CODE REVIEW

**Comment by caofangqi** [comment]
大佬你好   我使用的时候出现
com.mysql.jdbc.exceptions.jdbc4.MySQLNonTransientConnectionException: Can't call rollback when autocommit=true
导致无法提交，设置数据库自动提交关闭之后 还是这个错  求大佬指点

**Comment by caofangqi** [comment]
亲测可用，感谢大佬

**Comment by jorrellz** [comment]
-  Applicable for MySQL only: Add &relaxAutoCommit=true to the end of your JDBC connection string / URL.
Example:
 - jdbc:mysql://localhost/bamdb?autoReconnect=true&relaxAutoCommit=true

@caofangqi

**Comment by caofangqi** [comment]
@jorrellz 我去百度了下这个relaxAutoCommit参数的解释，如果驱动程序所连接的MySQL服务器的版本不支持事务，仍允许调用commit()、rollback()和setAutoCommit()？真／假，默认为“假”。  这个参数的作用是这样的吗，我有个疑问，就是如果MySQL服务器的版本不支持事务，调用commit()、rollback()和setAutoCommit()有用吗

File: tx-manager/src/main/java/com/codingapi/tm/manager/service/impl/MicroServiceImpl.java [hunk 1]
```diff
@@ -48,7 +50,7 @@ private boolean isIp(String ipAddress) {
     @Override
     public TxState getState() {
         TxState state = new TxState();
-        String ipAddress = discoveryClient.getLocalServiceInstance().getHost();
+        String ipAddress = registration.getHost();
```
Comment (YangAnLin) [inline tx-manager/src/main/java/com/codingapi/tm/manager/service/impl/MicroServiceImpl.java]: getHost  找不到这个方法

Comment (YangAnLin) [reply tx-manager/src/main/java/com/codingapi/tm/manager/service/impl/MicroServiceImpl.java]: registration.getHost(); 这个找不到

**Comment by jorrellz** [comment]
Check the database storage engine type，If type is MyISAM，Change to InnoDB.
@caofangqi

**Comment by caofangqi** [comment]
好的，谢谢你哈 
 @jorrellz

**Comment by standinhell** [comment]
> 1. Update SpringBoot version to 2.0.4.RELEASE
> 2. Update JDK version to 1.8
> 3. Update Spring Cloud version  to Finchley.RELEASE

想请教大佬，我使用这个版本报错

java.net.ConnectException: Connection refused: no further information
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:1.8.0_181]
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[na:1.8.0_181]
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:357) ~[zookeeper-3.5.3-beta.jar:3.5.3-beta-8ce24f9e675cbefffb8f21a47e06b42864475a60]
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1214) ~[zookeeper-3.5.3-beta.jar:3.5.3-beta-8ce24f9e675cbefffb8f21a47e06b42864475a60]

请各位大佬指点

**Comment by YangAnLin** [comment]
官网提供的 客户端demo,升级到2.0.3 就拍不起来,`Error creating bean with name 'serverListener': Lookup method resolution failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [com.codingapi.tx.springcloud.listener.ServerListener] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]`,这个这个怎么解决

**Comment by YangAnLin** [comment]
官网提供的 客户端demo,升级到2.0.3 就拍不起来,`Error creating bean with name 'serverListener': Lookup method resolution failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [com.codingapi.tx.springcloud.listener.ServerListener] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]`,这个这个怎么解决

**Comment by xynlanlan** [comment]
java.util.concurrent.RejectedExecutionException: Task com.codingapi.tx.netty.handler.TransactionHandler$1@3a97aac5 rejected from java.util.concurrent.ThreadPoolExecutor@451dbcc3[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 21]

线程超出最大值了

**Pull request closed without merging**